### ISSUE TITLE
False positive in `mocha/max-top-level-suites`

### ISSUE THREAD

**Issue by zachlysobey** [lo1tuma/eslint-plugin-mocha#224]
I noticed that the `mocha/max-top-level-suites` rule seems to trigger erroneously if there is a function which generates nested describes *outside* of the scope of the top-level-suite.

```js
describe("top-level describe", () => {
  it("has some tests");
  addNestedDescribe();
});

function addNestedDescribe() {
  describe("added nested describe", () => {
    it("has more tests");
  });
}
```

This creates the following output (notice, only one top-level describe)

```
top-level describe
- has some tests
  added nested describe
  - has more tests
```

But the rule still causes an error.

```
reproduction-of-bug.spec.js
  7:5  error    The number of top-level suites is more than 1  mocha/max-top-level-suites
```

It seems that this can be resolved in my case by moving that function *inside* of the top-level describe callback, but I would generally prefer not to have to do that.

**Comment by lo1tuma** [comment]
Thanks for reporting. Looks like a bug to me.

**Comment by zachlysobey** [comment]
I should note, that I'd be willing to help out with this. I'm not sure if I have the AST chops to fix it, but I could certainly write a failing unit test, or add a note to the documentation.
